lms/djangoapps/letstudy_api/views.py

```diff
@@ -0,0 +1,127 @@
+"""
+API end-point for creating a new user without password.
+"""
+
+from uuid import uuid4
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from social_django.models import UserSocialAuth
+
+from enrollment.api import add_enrollment, get_enrollment, update_enrollment
+from enrollment.errors import CourseModeNotFoundError
+from enrollment.views import ApiKeyPermissionMixIn
+from lms.djangoapps.instructor.views.tools import get_student_from_identifier
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from openedx.core.lib.exceptions import CourseNotFoundError
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from student.models import UserProfile
+from third_party_auth.models import OAuth2ProviderConfig
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    """
+    API endpoint for user provisioning.
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Create user by email, username and uid.
+        """
+        data = request.data
+        email = data.get('email')
+        username = data.get('username')
+        uid = data.get('uid')
+        first_name = data.get('first_name')
+        last_name = data.get('last_name')
+        full_name = "{} {}".format(first_name, last_name).strip() if first_name or last_name else username
```

Comment:
That construction works incorrectly in the case when one param exists and another is None. Please improve.

Reply:
I would format full_name in next way:
```
full_name = "{} {}".format(first_name, last_name).strip() or username
```